src/Stratis.Bitcoin/Controllers/NodeController.cs

```diff
@@ -327,45 +327,51 @@ public IActionResult DecodeRawTransaction([FromBody] DecodeRawTransactionModel r
         [HttpGet]
         public IActionResult ValidateAddress([FromQuery] string address)
         {
-            try
+            Guard.NotEmpty(address, nameof(address));
+
+            var result = new ValidatedAddress
             {
-                Guard.NotEmpty(address, nameof(address));
+                IsValid = false,
+                Address = address,
+            };
 
-                var res = new ValidatedAddress
-                {
-                    IsValid = false
-                };
+            try
+            {
                 // P2WPKH
                 if (BitcoinWitPubKeyAddress.IsValid(address, this.network, out Exception _))
                 {
-                    res.IsValid = true;
+                    result.IsValid = true;
                 }
-
                 // P2WSH
                 else if (BitcoinWitScriptAddress.IsValid(address, this.network, out Exception _))
                 {
-                    res.IsValid = true;
+                    result.IsValid = true;
                 }
-
                 // P2PKH
                 else if (BitcoinPubKeyAddress.IsValid(address, this.network))
                 {
-                    res.IsValid = true;
+                    result.IsValid = true;
                 }
-
                 // P2SH
                 else if (BitcoinScriptAddress.IsValid(address, this.network))
                 {
-                    res.IsValid = true;
+                    result.IsValid = true;
+                    result.IsScript = true;
                 }
-
-                return this.Json(res);
             }
-            catch (Exception e)
```

Comment:
Why not keep also the regular catch? that will return `ErrorHelpers.BuildErrorResponse`

Reply:
Good idea, ill put that in.

Reply:
Done.